spec/integration/application/filebucket_spec.rb

```diff
@@ -70,10 +70,114 @@
         filebucket.command_line.args << 'fac251367c9e083c6b1f0f3181'
         filebucket.run
       }.to output(a_string_matching(
-        %r{Debug: HTTP GET https:\/\/127.0.0.1:#{port}\/puppet\/v3\/file_bucket_file\/md5\/fac251367c9e083c6b1f0f3181 returned 200 OK}
+        %r{Debug: HTTP GET https:\/\/127.0.0.1:#{port}\/puppet\/v3\/file_bucket_file\/md5\/fac251367c9e083c6b1f0f3181\?environment\=production returned 200 OK}
       ).and matching(
         %r{something to store}
        )).to_stdout
     end
   end
+
+  context 'diff', unless: Puppet::Util::Platform.windows? || Puppet::Util::Platform.jruby? do
```

Comment:
I also never added a test to ensure it exits gracefully. I don't think it did before. I can dig more into that today if you think it's important though

Reply:
I think it's good as is